### ISSUE TITLE
Need more control over showDialogOrMenu() in case of dialog.

### ISSUE THREAD

**Issue by slavap** [lcuis/search_choices#84]
In current implementation: if (widget.dialogBox) { await showDialog( 
There are two problems:
1. I cannot pass all needed params to showDialog() call, e.g. anchorPoint and barrierDismissible.
2. I want to replace showDialog() completely by my custom show dialog functionality, so code should be refactored to allow custom dialog substitution, i.e. "clean" callback function with all needed data passed to construct dialog and returns selectedResult.

**Comment by lcuis** [comment]
Thanks for the improvement suggestion @slavap !

**Comment by lcuis** [comment]
Hello @slavap ,

I exposed the parameter `showDialogFn` in release 2.1.0 . This is not exactly what you asked for but this seems to cover your main needs. Can you please let me know whether that's fine or not?

**Comment by slavap** [comment]
@lcuis Yes, it covers my main goal, I can substitute needed params to showDialog() and even call my own dialog functionality. Thank you! 
Though additional refactoring to allow more control over menuWidget would be nice, because currently I still have to use menuWidget(searchTerms: searchTerms) as a "black box" without any control over selection logic or selection widget (though priority of that is not so urgent).

**Comment by lcuis** [comment]
Thanks for the confirmation and the secondary need clarification @slavap .
I will thus keep this issue open for now.
Feel free to propose a signature for the menu widget callback or even a PR if you'd like. I'll be happy to review and update the example and integration tests accordingly.
